### ISSUE TITLE
Forgetting "use strict" results in the entire file being underlined

### ISSUE THREAD

**Issue by paulsmithkc** [eslint/eslint#14782]
### Overview

With the rule configuration `strict: ['warn', 'global']`

If you forget to include the `'use strict'` directive at top of the file, it highlights the entire file, rather than just the first line.

![image](https://user-images.githubusercontent.com/6501079/124312495-07783480-db35-11eb-93b2-9699628aea74.png)

Expected behavior would be to only show the warning on the first line.

**What rule do you want to change?**

strict

**Does this change cause the rule to produce more or fewer warnings?**

Yes

**How will the change be implemented? (New option, new default behavior, etc.)?**

new option

**Please provide some example code that this change will affect:**

With the rule configuration `strict: ['warn', 'global']`

```js
let x = 1;
let y = 2;
let z = 3;
```

**What does the rule currently do for this code?**

The entire file is underlined. Making it difficult to see/count other issues.

**What will the rule do after it's changed?**

It will only underline the first line.

**Are you willing to submit a pull request to implement this change?**

Yes.

**Comment by aladdin-add** [comment]
a good enhancement to me. I will support it!

**Comment by bmish** [comment]
I don't think we need a new option to control where the violation is reported...

**Comment by nzakas** [comment]
We don't need a new option, however, I'd argue that underlining the entire file is the correct behavior because strict mode applies to the entire file.

**Comment by paulsmithkc** [comment]
@nzakas If the rules classify it as an error. I totally agree. However, when it is classified as a warning, that indicates fixing it is a "should" rather than a must.

So I think there are definitely arguments for highlighting the entire file in most cases. But I also see the need for an escape hatch when the rules are less strict.

Hence why I'd propose that the default behavior should stay for errors at least. And maybe warnings only flag the first line? Or adding another config option to make this an explicit opt-in.

**Comment by nzakas** [comment]
The rule doesn't know if its configured as an error or a warning, and that's by design, because we don't want rules to behave differently depending on how they are configured.

Given that it seems there's agreement that the current behavior is mostly correct, I don't see us making a change here.
